design/ironic-debuggability-improvement.md

```diff
@@ -0,0 +1,108 @@
+<!--
+ This work is licensed under a Creative Commons Attribution 3.0
+ Unported License.
+
+ http://creativecommons.org/licenses/by/3.0/legalcode
+-->
+
+# ironic-debuggability-improvement
+
+## Status
+
+provisional
+
+## Summary
+
+Ironic and Ironic-inspector logs are currently hidden inside their respective
+containers and never shown to the end user by default.
+This document describes proposed changes in Ironic and Ironic-inspector
+logs handling to make debugging easier for the end user.
+
+## Motivation
+
+We need to make Ironic logs accessible to the end user.
+
+### Goals
+
+1. Make it easier to understand what happened when a deployment fails.
+2. Avoid old logs pile up and take all available space.
+
+### Non-Goals
+
+1. Modify Ironic logging events or log level.
+
+## Proposal
+
+The ironic-inspector-image includes a new script, ironic-inspection-log-watch,
+that can act as an entry point for a container to dump host inspection logs.
+
+The script should watch for Ironic host inspection log files to appear in
+"/shared/logs/ironic-inspector/ramdisk", wait for them to stop changing
+(to ensure the entire log is dumped), decompress them, print their contents
+with each line prefixed by the base file name, and then remove the file.
+
+The ironic-image includes a new script, ironic-provisioning-log-watch,
+that can act as an entry point for a container to dump host provisioning logs.
+
+The script should watch for Ironic host provisioning log files to appear in
+"/shared/log/ironic/deploy", wait for them to stop changing (to ensure the
+entire log is dumped), decompress them, print their contents with each line
+prefixed by the base file name, and then remove the file.
+
+The baremetal-operator repository contains a kustomize-based deployment for
+Metal3 services. That should be updated to include a container based on the
+ironic-inspector-image using the ironic-inspection-log-watch entry point to
+show the logs collected during inspection.That also should be updated to
+include a container based on the ironic-image using the
+ironic-provisioning-log-watch entry point to show the logs collected during
+deployment.
+
+## Design Details
+
+ironic-inspection-log-watch.sh will be created in
+`https://github.com/metal3-io/ironic-image`
+ironic-provisioning-log-watch.sh will be created in
+`https://github.com/metal3-io/ironic-inspector-image`
```

Comment:
This seems backwards.

Reply:
Good catch, thank you. I will swap these two once clear on the thread above.